crates/admin/src/cluster_controller/service/state.rs

```diff
@@ -410,15 +404,17 @@ impl TrimMode {
     /// Compute the safe trim points for each log, assuming that partitions are mapped to logs 1:1.
     /// For a given cluster state, determines the set of trim points for all partitions' logs.
     fn calculate_safe_trim_points(&self) -> BTreeMap<LogId, (Lsn, PartitionId)> {
+        let partition_table = Metadata::current().partition_table_ref();
```

Comment:
`Metadata::current()` will clone the Arc. You can get the inner ref instead with `Metadata::with_current(|m| m.partition_table_ref())`

Reply:
Updated, thank you!